### ISSUE TITLE
[0.6.3] LoadError: ArgumentError: ~user tilde expansion not yet implemented

### ISSUE THREAD

**Issue by cdluminate** [JuliaLang/julia#27958]
This failure occurs on Debian's buildd during test, on both [amd64](https://buildd.debian.org/status/fetch.php?pkg=julia&arch=amd64&ver=0.6.3-2&stamp=1530881736&raw=0) and [i386](https://buildd.debian.org/status/fetch.php?pkg=julia&arch=i386&ver=0.6.3-2&stamp=1530881969&raw=0).

However, I cannot reproduce this problem on my laptop and another package build service. May I ask for some advise?

FYI: https://buildd.debian.org/status/package.php?p=julia&suite=experimental

**Comment by ViralBShah** [comment]
```
    compile             |       88                        88
    distributed         |                           No tests
    [31;1mFAILURE[0m
Error in testset replcompletions:
Error During Test
  Got an exception of type LoadError outside of a @test
  LoadError: ArgumentError: ~user tilde expansion not yet implemented
  while loading /<<PKGBUILDDIR>>/test/replcompletions.jl, in expression starting on line 474
ERROR: LoadError: Test run finished with errors
while loading /<<PKGBUILDDIR>>/test/runtests.jl, in expression starting on line 29
make[2]: *** [Makefile:18: all] Error 1
make[2]: Leaving directory '/<<PKGBUILDDIR>>/test'
make[1]: *** [debian/rules:87: override_dh_auto_test] Error 2
make[1]: Leaving directory '/<<PKGBUILDDIR>>'
make: *** [debian/rules:75: build-indep] Error 2
```

**Comment by ViralBShah** [comment]
Off topic, Probably don't run on `mips`? We don't have a port for mips.

**Comment by CDLuminate** [comment]
I disabled the "replcompletions" test, and buildd successfully built the package on amd64 and i386. [detail](https://buildd.debian.org/status/package.php?p=julia&suite=experimental). BTW, we can just ignore the failures on architectures like mips*.

**Comment by garrison** [comment]
Does `$HOME` (i.e. `"/sbuild-nonexistent"`) actually exist on the build machine?  My guess is that the error results from [these lines](https://github.com/JuliaLang/julia/blob/d55cadc350d426a95fd967121ba77494d08364c8/test/replcompletions.jl#L549-L563), which attempt to create a directory called `tmpfoobar` in the user's home directory.  I would argue that messing with the user's home directory should be outside the scope of the tests, but I see no way to test this specific functionality without doing so.

Off topic: I think if a directory is going to be created in the user's home directory, it should at least have "julia" in the name somewhere and perhaps begin with an essentially random string of characters.  Right now the test would fail, I believe, if I already had a "~/tmpfoobar" (I've been known to name things like this!).

**Comment by CDLuminate** [comment]
@garrison By default, Debian's buildd will build package with user `sbuild` whose home is inexistent as you guessed. In order to prevent some other failures (e.g. `mkdir` failure), I have already set `HOME` to `/tmp`, see [here](https://salsa.debian.org/julia-team/julia/blob/master/debian/rules#L15) .

Each time the buildd builds a package, it would create a clean chroot environment and the `/tmp` directory should be clean. I followed the test code and found the lines you pointed out, however got no idea how it is broken.
